client/core/core.go

```diff
@@ -1052,11 +1052,35 @@ func (c *Core) connectedWallet(assetID uint32) (*xcWallet, error) {
 	return wallet, nil
 }
 
+// connectWallet connects to wallet and validates the known deposit address
+// after successful connection, if address found & it does not belong to
+// wallet, it generates new address then updates xcWallet and dbWallet,
+// therefore it might hold the wallet lock
 func (c *Core) connectWallet(w *xcWallet) error {
 	err := w.Connect(c.ctx)
 	if err != nil {
 		return codedError(connectWalletErr, err)
 	}
+	// If wallet has deposit address check if it's belongs to connected
+	// wallet.
+	// If found address doesn't belong to wallet generate new one.
+	addr := w.address
+	var mine bool
+	if addr != "" {
+		mine, err = w.OwnsAddress(addr)
+		if err != nil {
+			return err
+		}
+	}
+	if !mine && addr != "" {
```

Comment:
The second `if` should be nested in the first.

```go
if addr != "" {
	mine, err := w.OwnsAddress(addr)
	if err != nil {
		return err
	}
	if !mine {
		nAddr, err := c.newDepositAddress(w)
```

Reply:
opps